libraries/logalyzer.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+/**
+ * Analyze the log of a Chronos job using predefined keywords
+ */
+class Logalyzer_Library {
+    private $job;
+    private $system;
+    private $log;
+    private $warningKeys;
+    private $errorKeys;
+
+    /**
+     * @throws Exception
+     */
+    public function __construct($job) {
+        $this->job = $job;
+        $this->system = new System($this->job->getSystemId());
+        $path = UPLOADED_DATA_PATH . '/log/' . $job->getId() . '.log';
+        $log = Util::readFileContents($path);
+        if ($log === false) {
+            $this->log = "";
+        } else {
+            $this->log = $log;
+        }
+
+        // Grab keyword arrays at creation of this object. Changes during a job run make the result outdated, but consistent
+        $this->warningKeys = json_decode($this->system->getLogalyzerWarningKeywords());
+        $this->errorKeys = json_decode($this->system->getLogalyzerErrorKeywords());
+        $this->calculateAndSetHash();
```

Comment:
Do not update the Hash every time you initialize the Logalyzer Lib. Set hash when creating job (or even better when writing the first log line, thus when creating the log file) and after checking the entire file.